mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java

```diff
@@ -720,4 +720,22 @@ public void searchWithFullTextShouldReturnMailsWhenTextBodyMatchesAndNonContinuo
         assertThat(messageSearchIndex.search(session, mailbox, searchQuery))
             .containsExactly(2l, 3l);
     }
+
+    @Test
+    public void searchWithFullTextShouldReturnMailsWhenHtmlBodyMatches() throws Exception {
+        Assume.assumeTrue(messageSearchIndex.hasCapability(MessageSearchIndexCapabilities.Text));
+        SearchQuery searchQuery = new SearchQuery();
+        searchQuery.andCriteria(SearchQuery.textContains("Regarder"));
+        assertThat(messageSearchIndex.search(session, mailbox, searchQuery))
+            .containsExactly(7l);
+    }
+
+    @Test
+    public void searchWithFullTextShouldReturnMailsWhenHtmlBodyMatchesAndNonContinuousWords() throws Exception {
```

Comment:
and tests with :
- some 'a', 'the', 'of'
- matching like "regardons" for word "regarder"